### PR TITLE
Fix remote console for v4

### PR DESCRIPTION
**Pull request by masayag**
The remote console code is changed to support also v4.

https://bugzilla.redhat.com/show_bug.cgi?id=1512364

### CODE REVIEW

**Comment by masayag** [comment]
[Bug 1512364 - [RHV] Console access failed: undefined method `get_resource_by_ems_ref' for #<OvirtSDK4::Connection:0x000000133f3c80>](https://trello.com/c/1yTzNIwg/241-bug-1512364-rhv-console-access-failed-undefined-method-getresourcebyemsref-for-ovirtsdk4connection0x000000133f3c80)

**Comment by masayag** [comment]
@miq-bot add_label bug  gaprindashvili/yes

**Comment by masayag** [comment]
@miq-bot add_label bug
@miq-bot add_label gaprindashvili/yes

**Comment by oourfali** [comment]
@borod108 or @jhernand please review ASAP.

File: app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb [hunk 1]
```diff
@@ -204,6 +204,26 @@ def vm_reconfigure(vm, options = {})
       _log.info("#{log_header} Completed.")
     end
 
+    def remote_console_acquire_ticket(vm, userid, originating_server)
+      parsed_ticket = Nokogiri::XML(vm.with_provider_object(&:ticket))
```
Comment (borod108) [inline app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb]: Hi, was the old version not working for v3?
There seem to be changes compared to the code before the split.

Comment (masayag) [reply app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb]: It seems that we're not using the vm.provider_object anymore explicitly in our code base.

And due to the fact it carried the error with resolving the connection version, I'd like the idea of not using it if possible.

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit e80e45f6d8aa0f0e8442c57d32ff1969758fa98f
Author: Oved Ourfali <oourfali@redhat.com>
Date:   Thu Nov 16 10:20:38 2017 +0200

    Merge pull request #145 from masayag/support_remote_console_v4
    
    Fix remote console for v4
    (cherry picked from commit 27bc721af4a1db8fffce660d412acefa54b3217d)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1514137
```

**Pull request merged**